SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -147,6 +156,103 @@ void FLocalDeploymentManager::RefreshServiceStatus()
 	});
 }
 
+bool FLocalDeploymentManager::CheckIfPortIsBound(int32 Port)
+{
+	ISocketSubsystem* SocketSubsystem = ISocketSubsystem::Get(PLATFORM_SOCKETSUBSYSTEM);
+	bool bCanBindToPort = false;
+
+	// Set our broadcast address.
+	TSharedRef<FInternetAddr> BroadcastAddr = SocketSubsystem->CreateInternetAddr();
+	BroadcastAddr->SetBroadcastAddress();
+	BroadcastAddr->SetPort(Port);
+	
+	// Now the listen address.
+	TSharedRef<FInternetAddr> ListenAddr = SocketSubsystem->GetLocalBindAddr(*GLog);
+	ListenAddr->SetPort(Port);
+	const FString SocketName = TEXT("Runtime Port Test");
+	// Now create and set up our sockets.
+	FSocket* ListenSocket = SocketSubsystem->CreateSocket(NAME_Stream, SocketName, false /* bForceUDP */);
+	if (ListenSocket != nullptr)
+	{
+		ListenSocket->SetReuseAddr();
+		ListenSocket->SetNonBlocking();
+		ListenSocket->SetRecvErr();
+	
+		// Bind to our listen port.
+		if (ListenSocket->Bind(*ListenAddr))
+		{
+			bCanBindToPort = ListenSocket->Listen(0 /* MaxBacklog*/ );
+			ListenSocket->Close();
+		}
+		else
+		{
+			UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Failed to bind listen socket to addr (%s) for %s, the port is likely in use"), *ListenAddr->ToString(true), *SocketName);
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Failed to create listen socket for %s, the port is likely in use"), *SocketName);
+	}
+
+	// Either we couldn't create the socket or couldn't listen on it so the port is probably bound.
+	return !bCanBindToPort;
+}
+
+bool FLocalDeploymentManager::LocalDeploymentPreRunChecks()
+{
+	bool bSuccess = true;
+
+	// Check for the known runtime port (5301) which could be blocked.
+	if (CheckIfPortIsBound(RequiredRuntimePort))
+	{
+		// If it exists offer the user the ability to kill it.
+		if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("KillPortBlockingProcess", "A required port is blocked by another process (potentially by an old deployment). Would you like to kill this process?")) == EAppReturnType::Yes)
```

Comment:
Would it make sense to move the code inside `if (CheckIfPortIsBound(RequiredRuntimePort))` to some function `TryUnbindPort`?

Reply:
Good idea